File: ClientApp/src/app/excursion.ts
import { Sight } from './sight';
import { Tour } from './tour';
import { ExcursionSight } from './excursionsight';

export class Excursion {
  id: string;
  name: string;
  tours: Tour[];
  excursionSights: ExcursionSight[]
}

File: TourApi/TourApi/Models/Excursion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TourApi.Models
{
    public class Excursion
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public List<Tour> Tours { get; set; }
        public List<ExcursionSight> ExcursionSights { get; set; }
    }
}
